src/lib/storage/fixed_string_dictionary_segment.cpp

```diff
@@ -37,17 +38,22 @@ std::optional<T> FixedStringDictionarySegment<T>::get_typed_value(const ChunkOff
   DebugAssert(chunk_offset < size(), "ChunkOffset out of bounds.");
 
   const auto value_id = _decompressor->get(chunk_offset);
-  if (value_id == _dictionary->size()) {
+  if (value_id == _dictionary_span->size()) {
     return std::nullopt;
   }
-  return _dictionary->get_string_at(value_id);
+  return _dictionary_span->get_string_at(value_id);
 }
 
 template <typename T>
 std::shared_ptr<const FixedStringVector> FixedStringDictionarySegment<T>::fixed_string_dictionary() const {
   return _dictionary;
 }
 
+template <typename T>
```

Comment:
Maybe, we can delete the old method `fixed_string_dictionary`.

Reply:
Currently it's used inside the Hyrise Codebase.
We will remove it when the dictionary vectors get removed, but currently we don't want to touch too much parts of Hyrise at once.

Reply:
How and why should we get rid of it?

There are operators that access the dictionary prior to the segment (e.g., for early exists). Do you mean removing accesses to segment dictionaries at all?